SonarQube.Common/AnalysisConfig/AnalysisSetting.cs

```diff
@@ -27,5 +32,41 @@ public class AnalysisSetting
         /// </summary>
         [XmlAttribute]
         public string Value { get; set; }
+
+        #endregion
+
+        #region Static helper methods
+
+        /// <summary>
+        /// Regular expression to validate setting ids.
+        /// </summary>
+        /// <remarks>
+        /// Validation rules:
+        /// Must start with an alpanumeric character.
+        /// Can be followed by any number of alphanumeric characters or .
+        /// Whitespace is not allowed
+        /// </remarks>
+        private static readonly Regex ValidSettingKeyRegEx = new Regex(@"^\w[\w\d\.-]*$", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Returns true if the supplied string is a valid key for a sonar-XXX.properties file, otherwise false
+        /// </summary>
+        public static bool IsValidKey(string key)
```

Comment:
I moved the regular expression and supporting method from the WriteProjectInfoFile task here so that they can be shared. The properties file writer also validates that the keys are valid before writing them to the file.